### PR TITLE
Respect dns resolution from config

### PR DESCRIPTION
**Pull request by foxriver76**
for localhost and listen on all ip address

### CODE REVIEW

File: packages/cli/src/lib/setup/setupMultihost.ts [hunk 1]
```diff
@@ -99,12 +99,12 @@ export class Multihost {
...
             if (dbTools.isLocalObjectsDbServer(config.objects.type, config.objects.host, true)) {
                 console.log('Changing objects server to accept connections on all IP addresses.');
-                config.objects.host = '0.0.0.0';
+                config.objects.host = tools.getListenAllAddress();
                 changed = true;
             } else if (config.objects.type === 'redis') {
                 warningShown = true;
                 console.log(
-                    'Please check the binding of redis service. By default it is only local: http://download.redis.io/redis-stable/redis.conf\nChange "bind 127.0.0.1" to "bind 0.0.0.0" or to others.'
+                    `Please check the binding of redis service. By default it is only local: http://download.redis.io/redis-stable/redis.conf\nChange "bind ${tools.getLocalAddress()}" to "bind ${tools.getListenAllAddress()}" or to others.`
```
Comment (Apollon77) [inline packages/cli/src/lib/setup/setupMultihost.ts:107]: I think redis default is IPv4, so it should be always 127.0.0.1and not "specific" ... so the first change should be reverted

File: packages/cli/src/lib/setup/setupMultihost.ts [hunk 2]
```diff
@@ -114,12 +114,12 @@ export class Multihost {
...
             if (dbTools.isLocalStatesDbServer(config.states.type, config.states.host, true)) {
                 console.log('Changing states server to accept connections on all IP addresses.');
-                config.states.host = '0.0.0.0';
+                config.states.host = tools.getListenAllAddress();
                 changed = true;
             } else if (config.states.type === 'redis') {
                 !warningShown &&
                     console.log(
-                        'Please check the binding of redis service. By default it is only local: http://download.redis.io/redis-stable/redis.conf\nChange "bind 127.0.0.1" to "bind 0.0.0.0" or to others.'
+                        `Please check the binding of redis service. By default it is only local: http://download.redis.io/redis-stable/redis.conf\nChange "bind ${tools.getLocalAddress()}" to "bind ${tools.getListenAllAddress()}" or to others.`
```
Comment (Apollon77) [inline packages/cli/src/lib/setup/setupMultihost.ts:122]: some here for redis

File: packages/controller/src/main.ts [hunk 3]
```diff
@@ -1553,8 +1553,8 @@ function setIPs(ipList?: string[]): void {
 
     // check if IPs detected (because of DHCP delay)
     let found = false;
-    for (let a = 0; a < _ipList.length; a++) {
-        if (_ipList[a] === '127.0.0.1' || _ipList[a] === '::1/128' || !_ipList[a].match(/^\d+\.\d+\.\d+\.\d+$/)) {
+    for (const entry of _ipList) {
+        if (entry === '127.0.0.1' || entry === '::1/128' || !entry.match(/^\d+\.\d+\.\d+\.\d+$/)) {
```
Comment (Apollon77) [inline packages/controller/src/main.ts:1557]: die regex schliesst alle ipv6 aus??

Comment (foxriver76) [reply packages/controller/src/main.ts:1557]: good catch

**Comment by Apollon77** [comment]
But in fact to minimize edge cases I wpuld replace all the "===" cases by a "isLocalAddress(ip)"/"isListenAllAddress(ip)" which always checks for both variants independent of the setting. Else we might get too strange cases where someone changes the setting and then many logics go crazy because they do not detect local adress or such anymore because it also expects the config also to be changed directly (but config should not be changed before so needs to be after ...)

And then ideally export all 4 methods (get*/is*) via core to adapter devs. (and add them maybe to Readme at the place of the dns-lookup-setting config

File: packages/cli/src/lib/setup/setupMultihost.ts (hunk 1 above)
Comment (foxriver76) [reply packages/cli/src/lib/setup/setupMultihost.ts:107]: I made the assumption, that if someone wants to use ipv6 he has already configured the redis server accordingly, however we can change back

Comment (Apollon77) [reply packages/cli/src/lib/setup/setupMultihost.ts]: This is a good question ... idea was to give guidance for "default Redis configs" ... If someone touched that setting already he should also remember that

File: README.md [hunk 4]
```diff
@@ -565,6 +565,9 @@ Example: `"nodeProcessParams": ["--tls-min-v1.0"]`
 You can decide to use IPv6 first as DNS resolution method instead of IPv4 first (default). 
 To use this, configure `dnsResolution` to `verbatim` in `iobroker.json`.
 
+For handling IPv6 and IPv4 in adapters, we provide some convenience methods.
```
Comment (Apollon77) [inline README.md:568]: Please provide adapter-core version since when the methods are availabkle

**Review by Apollon77 — APPROVED** [review]
mini nit, but I approve already

**Pull request merged**